src/client.js

```diff
@@ -1,27 +1,56 @@
 "use strict";
 
-const net = require("net");
-const url = require("url");
+const net = require("net")
+const url = require("url")
+const {dialog} = require("electron")
 
 // Client can read/write messages from a TCP server
 class Client {
-  // init initializes the Client
-  init(addr) {
-    let u = url.parse("tcp://" + addr, false, false);
-    this.socket = new net.Socket();
-    this.socket.connect(u.port, u.hostname, function() {});
-    this.socket.on("close", function() {
-      process.exit();
-    });
-    return this;
-  }
-
-  // write writes an event to the server
-  write(targetID, eventName, payload) {
-    let data = { name: eventName, targetID: targetID };
-    if (typeof payload !== "undefined") Object.assign(data, payload);
-    this.socket.write(JSON.stringify(data) + "\n");
-  }
+    // init initializes the Client
+    init(addr) {
+        this.connect(addr)
+
+        this.socket.on('error', function(err){
+            // Raising an exception in case of any error in socket
+            const messageBoxOptions = {
+                type: "error",
+                title: "Error in Main process",
+                message: err
+            };
+            dialog.showMessageBox(messageBoxOptions);
+            process.exit(1);
+        });
+
+        this.socket.on('close', function() {
+            process.exit();
+        })
+        return this
+    }
+
+    // write writes an event to the server
+    write(targetID, eventName, payload) {
+        let data = {name: eventName, targetID: targetID}
+        if (typeof payload !== "undefined") Object.assign(data, payload)
+        this.socket.write(JSON.stringify(data) + "\n")
+    }
+
+    /*
+     * for proper socket closing, unix socket remains open even after
+     * quiting the application, this function ends the socket connection
+     */
+    close() {
+        this.socket.end();
+    }
+
+    connect(addr) {
+        if (net.isIP(addr) > 0) {
```

Comment:
I don't really like the idea of using `net.isIP` here.

Can you check whether simplifying to this works :

```js
this.socket = net.createConnection(addr);
```

If not, I'd rather see something along the lines of :

```js
let u = url.parse(addr, false, false);
if (u.protocol === "tcp:") {
  this.socket = new net.Socket();
  this.socket.connect(u.port, u.hostname, function() {});
} else {
  this.socket = net.createConnection(addr);
}
```

Reply:
I tried with [below code], but there was some issue in windows with this I'll try again and update the code accordingly.
```
this.socket = net.createConnection(addr);
```

Reply:
Pardon me for late response, I tried testing client code with below function it works very well with darwin and linux but somehow does not work in windows. The app gets installed but does not runs so debugging that is taking a little time. I did try the other approaches (i.e "tcp" check as you and the isIP() check) but somehow windows app is not working.

```
this.socket = net.createConnection(addr);
```

Below are the two approaches I have used to build the app, code snippet of bundler.json
```
"environments": [
  {"arch": "amd64", "os": "darwin"},
  {
    "arch": "amd64",
    "os": "windows",
    "env": {
      "CC": "x86_64-w64-mingw32-gcc",
      "CXX": "x86_64-w64-mingw32-g++",
      "CGO_ENABLED": "1"
    }
  }
]
```

and 

```
"environments": [
  {"arch": "amd64", "os": "linux"},
  {"arch": "386", "os": "windows"}
]
```

I need a bit more time to resolve this issue.